brick-editor.js

```diff
@@ -97,43 +99,39 @@ function buttonHandler(i) { // eslint-disable-line no-unused-vars
  *
  * The setTimeout() function is necessary to allow highlighting before confirm dialog popup
  *
+ * @param {string} buffer - The current editor text.
  * @param {AST} ast - The root of the ast to search through.
  * @param {Location} selection - An object with start/end lineNumber and start/end column properties.
  * @returns {undefined}
  */
-function selectionBranch(ast, selection) {
+function selectionBranch(buffer, ast, selection) {
```

Comment:
Never quite understand the naming of this function (or `blockBranch`, for that matter). Might be worth renaming.

Reply:
function names are hard. blockHandler and selectionHandler? it doesn't do the actual deleting and there's already a deleteBlock function.

Reply:
For now, I would actually get rid of these functions and merge them into the calling functions (onPointBackspace and onPointDelete for blockBranch; onRangeDelete for selectionBranch). I might later refactor onPointBackspace and onPointDelete to both call (for example) onPointRemove, but that can be a different PR.